From 7848ac716fd8b20e83cb7a0cd3344a396d6131e9 Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimo@endlessm.com>
Date: Thu, 24 Oct 2013 10:36:13 -0700
Subject: [PATCH] Fix finding the internal GNUChess copy

- configure.ac was not setting the GNUCHESS variable correctly
- we were not looking for the executable correctly in $PATH
---
 configure.ac                        | 2 +-
 src/chess_computer-activity/chess.c | 8 ++++++--
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index f645024..c9f456c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -356,7 +356,7 @@ else
   AC_DEFINE_UNQUOTED(SYSTEM_CONFIG_DIR, "/etc", [System GCompris config directory])
 
   dnl Test for gnuchess
-  if test x$GNUCHESS = xno; then
+  if test x$GNUCHESS = x; then
     GNUCHESS="gcompris-gnuchess"
   fi
   AC_DEFINE_UNQUOTED(GNUCHESS, "$GNUCHESS", Defines where GNU Chess resides on the system)
diff --git a/src/chess_computer-activity/chess.c b/src/chess_computer-activity/chess.c
index 7a3e4fa..761eeea 100644
--- a/src/chess_computer-activity/chess.c
+++ b/src/chess_computer-activity/chess.c
@@ -245,8 +245,12 @@ static void start_board (GcomprisBoard *agcomprisBoard)
 	{
 	  /* Check in our exec prefix */
 	  extern gchar *exec_prefix;
-	  gnuchess_bin = g_build_filename(exec_prefix, *gnuchess_pathptr,
-					  NULL);
+
+	  if (exec_prefix != NULL)
+	    gnuchess_bin = g_build_filename(exec_prefix, *gnuchess_pathptr,
+					    NULL);
+	  else
+	    gnuchess_bin = g_find_program_in_path(*gnuchess_pathptr);
 	}
 
       if (g_file_test (gnuchess_bin, G_FILE_TEST_IS_EXECUTABLE))
-- 
1.8.1.2

